crates/bevy_pbr/src/light_probe/environment_map.rs

```diff
@@ -0,0 +1,391 @@
+    /// map IDs to the index in `binding_index_to_cubemap`.
+    cubemap_to_binding_index: HashMap<EnvironmentMapIds, u32>,
+}
+
+/// A component, part of the render world, that stores the ID of the environment
+/// map attached to each view.
+///
+/// This is a simplified version of the structure used when binding arrays are
+/// not available on the current platform.
+#[cfg(any(feature = "shader_format_glsl", target_arch = "wasm32"))]
+#[derive(Component, Default)]
+pub struct RenderViewEnvironmentMaps {
+    /// The environment map attached to the view, if any.
+    cubemap: Option<EnvironmentMapIds>,
+}
+
+/// All the bind group entries necessary for PBR shaders to access the
+/// environment maps exposed to a view.
+#[cfg(all(not(feature = "shader_format_glsl"), not(target_arch = "wasm32")))]
+pub(crate) struct RenderViewBindGroupEntries<'a> {
+    /// A texture view of each diffuse cubemap, in the same order that they are
+    /// supplied to the view (i.e. in the same order as
+    /// `binding_index_to_cubemap` in [`RenderViewEnvironmentMaps`]).
+    ///
+    /// This is a vector of `wgpu::TextureView`s. But we don't want to import
+    /// `wgpu` in this crate, so we refer to it indirectly like this.
+    diffuse_texture_views: Vec<&'a <TextureView as std::ops::Deref>::Target>,
+
+    /// As above, but for specular cubemaps.
+    specular_texture_views: Vec<&'a <TextureView as std::ops::Deref>::Target>,
+
+    /// The sampler used to sample elements of both `diffuse_texture_views` and
+    /// `specular_texture_views`.
+    pub(crate) sampler: &'a Sampler,
+}
+
+/// All the bind group entries necessary for PBR shaders to access the
+/// environment maps exposed to a view.
+///
+/// This is the version used when binding arrays are not available on the
+/// current platform.
+#[cfg(any(feature = "shader_format_glsl", target_arch = "wasm32"))]
+pub(crate) struct RenderViewBindGroupEntries<'a> {
+    /// The texture view of the view's diffuse cubemap.
+    diffuse_texture_view: &'a TextureView,
+    /// The texture view of the view's specular cubemap.
+    specular_texture_view: &'a TextureView,
+    /// The sampler used to sample elements of both `diffuse_texture_view` and
+    /// `specular_texture_view`.
+    pub(crate) sampler: &'a Sampler,
+}
+
+impl ExtractInstance for EnvironmentMapIds {
+    type Data = Read<EnvironmentMapLight>;
+
+    type Filter = ();
+
+    fn extract(item: QueryItem<'_, Self::Data>) -> Option<Self> {
+        Some(EnvironmentMapIds {
+            diffuse: item.diffuse_map.id(),
+            specular: item.specular_map.id(),
+        })
+    }
+}
+
+impl RenderViewEnvironmentMaps {
+    pub(crate) fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[cfg(all(not(feature = "shader_format_glsl"), not(target_arch = "wasm32")))]
+impl RenderViewEnvironmentMaps {
+    /// Returns true if there are no environment maps for this view or false if
+    /// there are such environment maps.
```

Comment:
Doc comment wording is a bit confusing. You can just say whether there are any envmaps associated with the view, or something.